lib/src/server/interceptor.dart

```diff
@@ -0,0 +1,7 @@
+import '../shared/status.dart';
+
+import 'call.dart';
+
+abstract class Interceptor {
+  GrpcError handle(ServiceCall call);
```

Comment:
We might want to make this async, such that we can do general async handling in the interceptor

Reply:
Sounds interesting, looks like it will require a refactoring, old code is fully synchronous.
I tried to return Future<GrpcError> and mark functions async. UT started to fail.
@sigurdm Don't you mind to do it in separate PR?

Should Interceptor return always `Future<GrpcError>` in this case?